deployment/bin/entrypoint.sh

```diff
@@ -0,0 +1,96 @@
+#!/bin/bash
+#
+# This script is used as the entrypoint for the docker image
+# 
+# This entrypoint script defaults to using environment variables to populate
+# a jinja2 config template and writing out the config before starting the service
+#
+# If a first parameter is given it overrides the environment variables for the
+# data source and is either a file path or a URL
+# If a second parameter is given it overrides the default template and it should be
+# either a file path or a URL
+#
+# If there is a readable file at /run/secrets/auth_data then it will be read in
+# and passed as a header for a request to the data source URL, it must be in the
+# form "HEADER_NAME:VALUE" that httpie uses for custom headers. For example, to
+# set a gitlab private token header you would create a file that contains
+# "PRIVATE-TOKEN:mesoextrasecret"
+# A readable file at /run/secrets/auth_template is used for custom headers to the
+# template URL. Note that they are separated to avoid leaking creds to a URL that
+# doesn't require them
+#
+
+# Define an error exit function
+function error_exit
+{
+	echo "$1" 1>&2
+	exit 1
+}
+
+# This is the path to the jinja2 template tool, some derivative of
+# this: https://github.com/kolypto/j2cli
+export J2=/usr/local/bin/j2
+
+DIR="$( cd "$( dirname "$0" )" && pwd )"
+# Default config template
+TEMPLATE=$DIR/../conf/.templates/deployment.cfg.j2
+
+# Default data source is empty, resulting in env variables being used
+DATA_SRC=""
+
+# Set empty default values for the auth headers
+AUTH_DATA=""
+AUTH_TEMPLATE=""
+
+# Auth header secret paths - can be declared in docker-compose secrets declaration
+if [ -r "/run/secrets/auth_data" ]; then
+     AUTH_DATA=`cat /run/secrets/auth_data`
+fi
+if [ -r "/run/secrets/auth_template" ]; then
+    AUTH_TEMPLATE=`cat /run/secrets/auth_template`
+fi
+
+# If we have a first argument see if it is a file else treat a URL, this
+# is the dictionary context for the jinja2 template config file
+if [ "$1" ] ; then
+    if [ -r $1 ] ; then
+        DATA_SRC=$1
+    else
+        TMPDIR=/tmp/data$$
+        mkdir $TMPDIR
+        # Fetch the file, and have it error out on any redirects to avoid a 200 response
+        # that is just a redirect to a login screen
+        wget -q -nd --max-redirect=0 --no-check-certificate --header="${AUTH_DATA}" -P $TMPDIR -N $1  || \
```

Comment:
Why aren't we checking the certificate?

Reply:
Because we may be hitting an internal server that uses self-signed certs. We don't depend on the certs for server authentication, just to encrypt traffic so that our creds don't pass in the clear.

Reply:
Let's note that in a comment